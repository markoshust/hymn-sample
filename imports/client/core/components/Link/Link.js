import React, { PropTypes } from 'react';
import { compose, withHandlers } from 'recompose';

const Link = ({
  children,
  href,
  onClick,
}) => (
  <a
    href={href}
    onClick={onClick}
  >
    {children}
  </a>
);

Link.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
  id: PropTypes.string,
  onClick: PropTypes.func,
};

const enhance = compose(
  withHandlers({
    onClick: ({
      id,
      onClick,
    }) => (event) => {
      event.preventDefault();

      if (onClick) onClick(id);
    },
  })
);

export default enhance(Link);
