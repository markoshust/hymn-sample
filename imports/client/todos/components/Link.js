import React, { PropTypes } from 'react';
import { compose, withHandlers } from 'recompose';

const Link = ({ active, children, onClick }) => (
  active ? (
    <span>{children}</span>
  ) : (
    <a
      href="#"
      onClick={onClick}
    >
      {children}
    </a>
  )
);

Link.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

const enhance = compose(
  withHandlers({
    onClick: ({ onClick }) => (e) => {
      e.preventDefault();
      onClick();
    },
  })
);

export default enhance(Link);
