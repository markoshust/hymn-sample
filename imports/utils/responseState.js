const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
