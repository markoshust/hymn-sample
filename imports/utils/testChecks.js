import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
