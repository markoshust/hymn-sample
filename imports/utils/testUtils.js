import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
