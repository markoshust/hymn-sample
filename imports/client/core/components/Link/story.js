import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import Link from './Link';

const stories = storiesOf('core/Link', module);

stories.add('Standard', () => (
  <Link
    href="#"
    onClick={action('click')}
  >
    {text('Text', 'Some Text')}
  </Link>
));

stories.add('With ID', () => (
  <Link
    href="#"
    id={text('ID', 1)}
    onClick={action('click')}
  >
    {text('Text', 'Some Text')}
  </Link>
));
