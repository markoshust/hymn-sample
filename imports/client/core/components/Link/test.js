import React from 'react';
import { renderCheck } from 'utils/testChecks';
import Link from './Link';

const onClick = jest.fn();

describe('<Link />', () => {
  it('renders with text', () => {
    renderCheck(
      <Link>Some Text</Link>
    );
  });

  it('renders with text and onClick', () => {
    renderCheck(
      <Link onClick={onClick}>Some Text</Link>
    );
  });

  it('renders with text and onClick, id', () => {
    renderCheck(
      <Link
        id="abc"
        onClick={onClick}
      >
        Some Text
      </Link>
    );
  });
});
