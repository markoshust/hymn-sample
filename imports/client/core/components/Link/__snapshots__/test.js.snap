exports[`<Link /> renders with text 1`] = `
<a
  href={undefined}
  onClick={[Function anonymous]}>
  Some Text
</a>
`;

exports[`<Link /> renders with text and onClick 1`] = `
<a
  href={undefined}
  onClick={[Function anonymous]}>
  Some Text
</a>
`;

exports[`<Link /> renders with text and onClick, id 1`] = `
<a
  href={undefined}
  onClick={[Function anonymous]}>
  Some Text
</a>
`;
